beacon-chain/p2p/gossip_scoring_params.go

```diff
@@ -137,69 +151,96 @@ func defaultBlockTopicParams() *pubsub.TopicScoreParams {
+		FirstMessageDeliveriesWeight:    firstMessageWeight,
 		FirstMessageDeliveriesDecay:     scoreDecay(1 * oneEpochDuration()),
-		FirstMessageDeliveriesCap:       179,
-		MeshMessageDeliveriesWeight:     -0.064,
+		FirstMessageDeliveriesCap:       firstMessageCap,
+		MeshMessageDeliveriesWeight:     meshWeight,
 		MeshMessageDeliveriesDecay:      scoreDecay(1 * oneEpochDuration()),
-		MeshMessageDeliveriesCap:        float64(aggPerEpoch),
-		MeshMessageDeliveriesThreshold:  float64(aggPerEpoch / 50),
+		MeshMessageDeliveriesCap:        meshCap,
+		MeshMessageDeliveriesThreshold:  meshThreshold,
 		MeshMessageDeliveriesWindow:     2 * time.Second,
-		MeshMessageDeliveriesActivation: 32 * oneSlotDuration(),
-		MeshFailurePenaltyWeight:        -0.064,
+		MeshMessageDeliveriesActivation: 1 * oneEpochDuration(),
+		MeshFailurePenaltyWeight:        meshWeight,
 		MeshFailurePenaltyDecay:         scoreDecay(1 * oneEpochDuration()),
-		InvalidMessageDeliveriesWeight:  -140.4475,
+		InvalidMessageDeliveriesWeight:  -maxScore() / aggregateWeight,
 		InvalidMessageDeliveriesDecay:   scoreDecay(50 * oneEpochDuration()),
-	}
+	}, nil
 }
 
-func defaultAggregateSubnetTopicParams(activeValidators uint64) *pubsub.TopicScoreParams {
-	topicWeight := attestationTotalWeight / float64(params.BeaconNetworkConfig().AttestationSubnetCount)
-	subnetWeight := activeValidators / params.BeaconNetworkConfig().AttestationSubnetCount
-	minimumWeight := subnetWeight / 50
+func defaultAggregateSubnetTopicParams(activeValidators uint64) (*pubsub.TopicScoreParams, error) {
+	subnetCount := params.BeaconNetworkConfig().AttestationSubnetCount
+	// Get weight for each specific subnet.
+	topicWeight := attestationTotalWeight / float64(subnetCount)
+	subnetWeight := activeValidators / subnetCount
+	// Determine the amount of validators expected in a subnet in a single slot.
 	numPerSlot := time.Duration(subnetWeight / uint64(params.BeaconConfig().SlotsPerEpoch))
 	comsPerSlot := committeeCountPerSlot(activeValidators)
-	exceedsThreshold := comsPerSlot >= 2*params.BeaconNetworkConfig().AttestationSubnetCount/uint64(params.BeaconConfig().SlotsPerEpoch)
+	exceedsThreshold := comsPerSlot >= 2*subnetCount/uint64(params.BeaconConfig().SlotsPerEpoch)
 	firstDecay := time.Duration(1)
 	meshDecay := time.Duration(4)
 	if exceedsThreshold {
 		firstDecay = 4
 		meshDecay = 16
 	}
+	// Determine expected first deliveries based on the message rate.
+	firstMessageCap, err := decay(scoreDecay(firstDecay*oneEpochDuration()), float64(numPerSlot*2/gossipSubD))
+	if err != nil {
+		return nil, err
+	}
+	firstMessageWeight := maxFirstDeliveryScore / firstMessageCap
+	// Determine expected mesh deliveries based on message rate applied with a dampening factor.
+	meshThreshold, err := decayThreshold(scoreDecay(meshDecay*oneEpochDuration()), float64(numPerSlot)/dampeningFactor)
+	if err != nil {
+		return nil, err
+	}
+	meshWeight := -maxScore() / (aggregateWeight * meshThreshold * meshThreshold)
```

Comment:
Should we abstract this into a separate function, since the same calculation is happening in `defaultAggregateTopicParams()` above?